src/util/validation.h

```diff
@@ -6,12 +6,18 @@
 #ifndef BITCOIN_UTIL_VALIDATION_H
 #define BITCOIN_UTIL_VALIDATION_H
 
-#include <string>
+#include <consensus/validation.h>
 
-class ValidationState;
+#include <string>
 
 /** Convert ValidationState to a human-readable message for logging */
-std::string FormatStateMessage(const ValidationState &state);
+template <typename Result>
+std::string FormatStateMessage(const ValidationState<Result>& state)
```

Comment:
Could move this into ValidationState directly, and make it
```
std::string FormatStateMessage() const { return m_reject_reason + (m_debug_message.empty() ? "" : ", " + m_debug_message); }
```

Reply:
Done in 7239d01 but named it `ToString` since #16688 uses it for valid states as well.